New layout

andy013 opened the pull request:
Hey man, I had at go at redesigning the layout of the site. I'm not sure if you'll want to use it or not. I just did it for fun. There are some new cards and board images in here as well. You can use whatever you want.

A couple of notes. I added in a dependency called 'react-perfect-scrollbar'. It's a really nice scrollbar that's used in loads of big sites (facebook, discord etc.). The old scrollbar only worked in chrome. This one works in other browsers too. 

I am using 2 components of semantic-ui that weren't included in the local .css file. Popups and Flags. I just linked to the CDN semantic file in the head tag of layout-no-menu.pug but you'll probably want to rebuild the local version so that it includes these rather than duplicating lots of css (I wasn't sure how to do this, didn't want to mess anything up). Also there might be some other parts missing as including this file seemed to centre the modals properly. 

Currently emotes don't seem to be working for me, but these weren't working on master either so I assume it is work in progress.

I created a DevHelpers component quickly for testing but the default game button is currently broken. You can just delete it from App.jsx like before.

Also, I added a few props but forgot to add them to proptypes at the end of the files, not sure how important that is.

cozuya commented:
Awesome will take a look soon.  Master is not very usable right now (damn github pips).  You might want to branch off of the 0.8.2 stable branch.  Was not aware of problems with emotes or that the scrollbars don't work on non-chrome.. at least for prod, will check that out.  rc-slider has not been the best experience for me so would be nice to have a different slider.

I'm in a bit of dev hell as things keep coming up that I realize will break the game for people particularly with the new routing idea.. like just realized that anyone routing away from the game but not through the leavegame button will never leave the socket room and will keep getting yanked back.. fun.  So a bit backed up here and I'll take a look when I can.  If you want to get some discussion going a screenshot would be awesome.

andy013 commented:
Here's a quick screenshot of the main page:
![](https://i.imgur.com/igSdHnq.png)

cozuya commented:
Is this something you could make a player option/setting for?  If so it'd be very easy to get this merged in.

andy013 commented:
It would be quite a lot of work to do that tbh. I deleted/changed a bunch of the css from the old version, so I would need to go through and rename all the classes to something else and bring in the old stylesheet again before setting up some sort of conditional rendering. The problem is there will undoubtedly be shared stuff between both layouts so it will get quite tricky to prevent them interfering with each other. I don't think it's worth it. I think it would be easier to just put a poll up and see if people like it or change it and see what the feedback is and if people don't like it you can always revert back to the old layout.

cozuya commented:
Cool as I've said a million times I'm not a designer and there's really no reason to have a giant "defaultmid" section that does nothing.  Let me know when this pr is ready for review and I'll figure out a way to get it out to the community and see what people think.

andy013 commented:
Cool. It should be good to go now. There might be a few rough edges but I think everything is working.

andy013 commented:
I added an emote button to the chat now. I used the react version of semantic-ui as it was better than using jquery for the popups. Dependency is 'semantic-ui-react'.

andy013 commented:
I noticed a bug, generalchat is fine but gamechat is not adding spaces between words. I don't think it's anything I've added as I didn't touch those functions. I don't have time to look at it atm, just wanted to post it here so I don't forget.

cozuya commented:
0.9 mostly pushed up and will need MC resolution here.  going to put this up on the polls page in the patch.

cozuya commented:
I like it a ton so far.  Some bug with gamechat crashing the front end you'll see when you merge master IDK.  But looks great, way better than my crap.

andy013 commented:
Not sure what the bug is? Gamechat seems to work fine for me. I need to add some keys for array items but nothing else is showing in console. I did find an issue with rainbow games. Since they need to check wins and losses, if you refresh a rainbow game lobby while there are open seats then it will crash the front end because the player is removed from playerlist when they refresh so .wins is undefined.

cozuya commented:
yeah probably messed up the MC that I did quickly.  So got it up and working now.

Couple things: there's a gutter on the right in the "not-widescreen" lobby view I think you should get rid of/move the playerlist/genchat column over.

While I like the expanded descriptions in tracks/vote selection, I think they should not be there in experienced games.  I also really like my "spin in from the side" animations and would like them back tbh, right not its a little popup out of nowhere for me.

Definitely a few different key issues in the console with react, I can look into those if you'd like.

I'm not sure the "application width" player setting is working right but should maybe just get thrown out anyways.

Clicking on the logo should route to the home page imo.

Other than the above I love it!  I have taken some screenshots and will get the in front of the players/make a poll but I'd say why not, change is good!  I plan on 0.9 tomorrow morning.

andy013 commented:
Thanks for the feedback. I agree with everything you said. I am already working on trying to move the right side bar over a bit because it feels a bit too cramped right now. 

I could try and add a player option to disable those help messages into the settings.

I think the "spin in from the side" animation was deleted by accident. I didn't intentionally delete it. I'll have a look at master and see if I can bring it back.

cozuya commented:
Great I've made a poll and we'll see.  I like the idea a lot though especially like just scanning through the list can see who's playing where and all that just by cardbacks.  One more minor thing is that non-waiting-for-players games should probably not have the requirements to start are in them because they're already started.  If that doesn't make sense, in the SS below, the "fascists win" game has "5 / 5-10" on it when it should probably just be "5".  Also in 0.9 I made a "games list" header on the left sidebar and that might be helpful here i.e. some sort of "here's a list of games that are out there now" thing.

![image](https://user-images.githubusercontent.com/2840666/32140572-376c7724-bc34-11e7-8c25-d3f659f03218.png)

cozuya commented:
Early voting on your layout looks like 3:1 for/against btw so lets do it!  Let me know when its ready for final review.

andy013 commented:
Hey, this is ready for review. The main thing you will need to look at is the semantic stylesheets. In `layout-no-menu.pug` I moved the link to the CDN file outside the if loop. This means that even when we are in a game we are still loading this. The reason I done this is because I don't know how you built the `sematic.scss` file in `/src` and I didn't know how to add some components to it (flags, tooltips, responsivity / centring of modals). If you move the CDN link back into the if statement then you will see that some functionality breaks. E.g. mousing over emotes won't display a tooltip any more. The way it is currently works but is not ideal because lots of css is duplicated unnecessarily. It would be good if you could add the missing components into `semantic.scss` instead. 

I also removed the customWidth setting. I'm not sure if you need to delete these entries from the db as well.

I added most of the stuff you requested. I thought about removing the player requirement numbers after the game has start but I think it's better to just leave it as is. I don't like removing information that players may be interested in (e.g. I might want to scan down the list and see what preferences other players have) and I don't think it's confusing as there are other distinct visual changes to signal to the player that a game is in progress (e.g. seat icons disappear, border changes color).

I added my game name to the contributors list, in case you wonder who Wi1son is.  

A bit of feedback: I really think that private games should be visible on the games list like they were before. I honestly don't think anyone is having a private conversation in the game chat and it makes it more convenient for some people if they can just quickly scan the list rather than copying a link and sending it to everyone. I saw multiple people confused as to how they could play with their friends after the update. You could always add a filter to the gameslist to show or hide private games if you felt they were cluttering it.

cozuya commented:
Yeah can't remember what/how I did with the semantic one, if by still loading you mean there's a slight delay on page load thats fine - if there's missing stuff though thats not good I'll look into it.

NP re: customwidth and I don't care about "dangling" db entries like that.  If you didn't pull it out of the model I will.

Sounds good re: player requirements.

Private game stuff hmm.  I haven't had a lot of feedback on that yet.  So my instructions to mods are pretty clear about not wanting hate speech (n word) in public chat.  But tbh if someone is playing a private game (which now no longer counts towards winrate anyways) I don't really care what they say to their friends.  I think a better solution would be to revert the private game gamelist change, but then to not be able to see chat at all in it unless you're seated or a mod.  But after remake game feature goes in it will be easier to play private games.  IDK will ask for more opinions.

Regardless I really want remake games done as I promised it was coming anyways so going to finish that up in the next few days and will review this for the next release!

andy013 commented:
Okay, I just removed semantic.scss for now. Hopefully everything should be included in the CDN version.

cozuya commented:
k now that 0.9.1 is out I'm going to look to start review here.

andy013 commented:
Sorry if you tried this and it didn't work for a while. I was trying to merge my changes into an older branch and for some reason github kept changing this branch as well.

cozuya commented:
Hey I'll poke it some more but I don't seem to get responses to both signup or signin somehow? hmm.  Did npm i.  I did move to the newest LTS but can't imagine thats the problem.

andy013 commented:
Hmm, its working okay for me and I didn't touch any of that code so I'm not sure what it could it be.

cozuya commented:
Weird yeah no change to accounts.js so uh will have to keep trying things.

andy013 commented:
Might be the change to app.js? I removed `${process.env.MONGOPORT}` to get it working for me.

cozuya commented:
That was it!  thanks mongo for your silent fails with no errors.

cozuya commented:
Poked at this a bit, I like it a lot, only some minor issues, if you want to tackle them awesome, if not I can.

- I think default cardback is a bit heavy contrast, if like much lighter red or ~0.2 opacity for the lizard outline would be better/make more sense.
- I'd like a tooltip when someone hovers over a game in the list and the arrow signifying "enter this game" appears.
- Still a right gutter at large widths.
- "Election #0" in the game list while a game starts is a bit clunky, "Game is starting" probably works better.
- Not sure the blue logout/login/signup button colors really match the top nav color well.
- If you wanted to go nuts, the box-shadow around a player in the default/lobby view should reflect their player color.

Other than that it looks great, players will love it.  You have a lot of design talent that I don't have so thats awesome.  Thanks and will look to pull this into master this weekend most likely and get this out in 0.10.0.

andy013 commented:
I might need an extra few days to get this wrapped up. The routing changes have some weird edge cases I'm trying to figure out.

cozuya commented:
No problem!  Take your time.

andy013 commented:
I opened another PR for this.
